src/js/airplay/AirPlayManager.ts

```diff
@@ -0,0 +1,210 @@
+ * - Open the AirPlay device picker via `prompt`, preferring WebKit
+ *   AirPlay on Apple platforms and falling back to the Remote Playback API
+ *   where available.
+ * - Listen to low-level API events and re-emit them as high-level
+ *   `EVENTS.AIRPLAY.*` events on the Video.js player.
+ */
+export class AirPlayManager {
+   private readonly _player: VideoJsPlayer;
+   private _remotePlayback: RemotePlayback | null = null;
+   private _webkitAirPlaySupported = false;
+
+   public constructor(player: VideoJsPlayer) {
+      this._player = player;
+      this._initialize();
+   }
+
+   public async isAirPlayAvailable(): Promise<boolean> {
+      if (!checkClientAirPlaySupport()) {
+         videojs.log(LOG_MESSAGES.AIRPLAY_NOT_SUPPORTED);
+         return false;
+      }
+
+      videojs.log(LOG_MESSAGES.AIRPLAY_SUPPORTED);
+
+      if (this._remotePlayback) {
+         try {
+            await this._remotePlayback.watchAvailability((available: boolean) => {
+               this._player.trigger(EVENTS.AIRPLAY.AVAILABILITY_CHANGE, { available });
+            });
+            return true;
+         } catch(error) {
+            videojs.log.error('Error checking Remote Playback availability', error);
+         }
+      }
+
+      if (this._webkitAirPlaySupported) {
+         return this._checkWebKitAirPlayAvailability();
+      }
+
+      return false;
+   }
+
+   public getState(): RemotePlaybackState | null {
+      return this._remotePlayback?.state || null;
+   }
+
+   public isConnected(): boolean {
+      return this._remotePlayback?.state === 'connected';
+   }
+
+   /*
+    * Open the AirPlay device picker for the current media element.
+    *
+    * This prefers the WebKit AirPlay API (Safari / Apple devices) when available,
+    * falling back to the standardized Remote Playback API for limited
+    * cross-browser support. If neither API is available, an error is thrown.
+    */
+   public async prompt(): Promise<void> {
+      if (this._webkitAirPlaySupported) {
+         const videoElement = getMediaElement<HTMLVideoElementWithAirPlay>(this._player);
+
+         if (!videoElement) {
+            videojs.log.error('Video element not found');
+            return;
+         }
+
+         if ('webkitShowPlaybackTargetPicker' in videoElement) {
+            videojs.log(LOG_MESSAGES.WEBKIT_AIRPLAY_PREFERRED);
+            videoElement.webkitShowPlaybackTargetPicker();
+            return Promise.resolve();
+         }
+      }
+
+      if (this._remotePlayback) {
+         videojs.log('Using Remote Playback API - limited device compatibility');
+         return this._remotePlayback.prompt();
+      }
+
+      throw new Error('No AirPlay API available');
+   }
+
+   private _initialize(): void {
+      const videoElement = getMediaElement(this._player);
+
+      if (!videoElement) {
+         videojs.log.error('Video element not found');
+         return;
+      }
+
+      if (checkClientAirPlaySupport() && 'remote' in videoElement) {
```

Comment:
If we had a type guard for `HTMLVideoElement`, we would not need this ad-hoc check.